### DISCUSSION TITLE
Save recon-settings as part of image meta info?

### DISCUSSION THREAD

**Discussion by KrisThielemans** [UCL/STIR#846] [Ideas]
In many cases it would be useful to save the info how an image was reconstructed (suggested by @zeljkozeljko). This is in general probably hard, but in most cases, STIR's `recon.parameter_info()` will give 99% of the info, so we could start there.

Some things to think about:
- for iterative algos, you would want to know the subiteration number. This isn't included in `recon.parameter_info()`
- this would have to be stored in some meta_data member of `DiscretisedDensity`. Do we create a constructor argument? (could become messy, so I suggest to have only `set_recon_info` member.
- ideally it would be updated with other post-processing (filtering, whatever) but this seems to become very hard
- no file format allows us to store this. We can create an Interfile key though (seems very easy with `add_parsing_key`).

In summary, a basic version seems fairly easy, while a foolproof version would be very hard. The question is therefore if it's beneficial. Any opinions?

**Comment by robbietuk** [comment]
In my own code I output a json `records`/`diagnostic` file that contains all information relavent to the reconstructions. This includes all the reconstruction information and paths to each of the other files, e.g. images output by the algorithms and initialisation parameters. I say this because rather than keeping the meta data in each of the images, I think an additional output file from each iterative algorithm containing all the meta data might be more useful.
